File: scala/80.scala
object Solution {
    def removeDuplicates(nums: Array[Int]): Int = {
        var last = nums.head
        var ptr = -1
        var count = 0
        nums.foreach(n => {
            if (last == n){
                count += 1
                if (count <= 2) {
                    ptr += 1
                }
            }
            else {
                ptr += 1
                count = 1
            }
            last = n
            nums(ptr) = n
        })
        ptr+1
    }
}

File: leetcodeJava/80.java
package leetcodeJava;

class Solution {
    public int removeDuplicates(int[] nums) {
        var l = 0; 
        for (var i = 0; i < nums.length; i++) {
            var count = 0;
            for (var j = i; j <= nums.length; j++) {
                if (j < nums.length && nums[j] == nums[i]) {
                    count++;
                } else {
                    i = j-1;
                    break;
                }
                if (count <= 2) {
                    nums[l] = nums[i];
                    l++;
                }
            }
        }
        return l;
    }
}